src/app/edit-movie-detail/edit-movie-detail.component.ts

```diff
@@ -14,61 +12,34 @@ export class EditMovieDetailComponent implements OnInit {
 
   @Input()
   movieId: string;
-  /*
-  movieForm: FormGroup;
-    id: '';
-    title: '';
-    posterUrl: string = '';
-    genre: string = '';
-    rating: number = null;  */
-
-    movieForm: FormGroup
 
-  constructor(private router: Router, private route: ActivatedRoute, 
-              private movieService: MovieService, private formBuilder: FormBuilder,
-              private activeModal: NgbActiveModal) {
+  form: FormGroup;
 
-                this.movieForm = this.formBuilder.group({
-                  title: [''],
-                  posterUrl: [''],
-                  genre: [''],
-                  rating:['']
-                  })
-              }
+  constructor(
+    private movieService: MovieService,
+    private formBuilder: FormBuilder,
+    private activeModal: NgbActiveModal
+  ) {
 
-  ngOnInit(): void {
-    console.log(this.movieId);
-    this.getMovieById(this.movieId);
-    this.movieForm = this.formBuilder.group({
-      id: [null, Validators.required],
-      title: [null, Validators.required],
-      posterUrl: [null, Validators.required],
-      genre: [null, Validators.required],
-      rating:[null, Validators.required]
+    this.form = this.formBuilder.group({
+      title: ['', [Validators.required]],
+      posterUrl: ['', [Validators.required]],
+      genre: ['', [Validators.required]]
     });
   }
 
-  getMovieById(id: any){
-    this.movieService.getMovieById(id).subscribe((data: any)=>{
-      this.movieForm.setValue({
-        id: data._id,
-        title: data.title,
-        genre: data.genre,
-        rating: data.rating,
-        posterUrl: data.posterUrl
-      });
-    });
+  ngOnInit(): void {
+    this.populatFormWithMovieData(this.movieId);
   }
 
-  onFormSubmit(){
-    this.movieService.updateMovie(this.movieId, this.movieForm.value)
-    .subscribe((res: any) => {
-      this.activeModal.close();
-      location.reload()
-      /*this.movieService.getMovies().subscribe(movies =>{
-        this.movieService.movieList = movies;
-      });*/
-    });
+  async populatFormWithMovieData(id: string) {
```

Comment:
Think we have typo error..should be "populateFormWithMovieData" ...

Reply:
Spot on! Let's try fixing this as an exercise to refresh Git topics.